### DISCUSSION TITLE
Sign In with Yahoo

### DISCUSSION THREAD

**Discussion by BlindDev** [invertase/react-native-firebase#8201] [Q&A] [answered]
Hey guys, 
Is there any example of adding Yahoo sign as social login option?

I couldn't find a way to accordingly use `signInWithPopup`. And the `react-native-app-auth` also doesn't have it in the tested providers list.

I managed to build Yahoo authentication using their API and got the access token, but now I'm stuck building an appropriate `credential` for the `signInWithCredential`. I can probably use `react-native-app-auth` instead of my API, but then not sure how to use the result with rn-firebase.

Thanks in advance! 🙏

**Comment by mikehardy** [comment, marked as answer]
Interesting - if you have a valid auth token from yahoo, you should be able to use the native oauth provider *through* here I think, and I think the Github sign in provider might be the best example - it should be...pretty close to how you do it ? It is going to require code here though - a new javascript-level provider and then logic to use it (implemented in the modular directory - the non-modular APIs are deprecated and may be ignored). Then adding recognition of the provider in the various native switch statements that load the specific native provider based on the javascript-level provider, so it loads the Yahoo one in the native firebase library


provider example https://github.com/invertase/react-native-firebase/blob/main/packages/auth/lib/providers/GithubAuthProvider.js

switch in android code (it's really just "add the import for the firebase provider, add an entry in switch statement for it" I think

https://github.com/invertase/react-native-firebase/blob/main/packages/auth/android/src/main/java/io/invertase/firebase/auth/ReactNativeFirebaseAuthModule.java#L1876

switch in ios code 

https://github.com/invertase/react-native-firebase/blob/main/packages/auth/ios/RNFBAuth/RNFBAuthModule.m#L1390-L1391

I wouldn't be intimidated by this at all vs successfully getting an auth token from yahoo. I always find that stuff to be pretty hard but while I can't guarantee it I think adding the Yahoo provider at the javascript layer then the appropriate switch entries in native code is...I hope easy

**Reply by BlindDev** [reply]
@mikehardy

It fails with `INVALID_CREDENTIAL_OR_PROVIDER_ID`

And I assume that this is similar to Microsoft social login limitation mentioned in the rn-firebase [docs](https://rnfirebase.io/auth/social-auth#microsoft)

I see exact similar note for manual sign-in flow.

Do you think I can try the same approach as for Microsoft?

**Reply by mikehardy** [reply]
Hmm I wish I could tell you with certainty but I think you are first here. I don't think there's a way to know without trying. All I know with certainty is that fir some reason yahoo is still quite popular as an email (and thus auth) provider, and that adding auth providers us usually just a tiny amount of code here in this module. Hopefully that's still the case here and if you follow the Microsoft auth example it works

**Reply by BlindDev** [reply]
Copy that. Will play with it next year. Thank you. Happy Holidays!
